_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,143 @@
+
+## Test Cases
+
+### Passed
+
+#### Pass example 1
+
+The `<svg>` element has a role of `img` which takes it's accessible name from it's `<title>` element.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100">
+  <title>A yellow circle</title>
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 2
+
+The `<svg>` element has a role of `img` and an `aria-label` containing the accessible name.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100" aria-label="A yellow circle">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 3
+
+The `<svg>` element has the role of `img` which takes it's accessible name from it's child content, the `<text>` element.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+  	<text font-size="36" fill="black" font-family="Verdana" text-anchor="middle" alignment-baseline="middle" x="22" y="34.4">A yellow circle</text>
+</svg>
+```
+
+#### Pass example 4
+
+The `<svg>` element has the role of `graphics-document` which takes it's accessible name from it's child content, the `<text>` element.
+
+```html
+
+<svg xmlns="http://www.w3.org/2000/svg"
+     xmlns:xlink="http://www.w3.org/1999/xlink" role="graphics-document">
+    <title>3 circles</title>
+    <desc>3 yellow circles with a green stroke horizontally adjacent to each other
+    </desc>
+    <g role="group">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="150" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="250" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+    </g>
+</svg>
+```
+
+### Failed
+
+#### Failed example 1
+
+The `<svg>` element has a rle of `img` but has no accessible name.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 2
+
+The `<svg>` element has an empty `title`.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" role="img">
+	<title> </title>
+  	<circle role="none" cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+The `svg` element does nott have a role of `img` or `graphics-document`. 
+
+```html
+<svg width="100" height="100">
+<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Inapplicable example 2
+
+The `svg` element is not included in the accessibility tree.
+
+```html
+<svg width="100" height="100">
+<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
```

Comment:
To demonstrate the multiple case where the svg can be or not included, I suggest an another inapplicable example : 

````html
#### Inapplicable example 3
 
 The `svg` is not included in the accessibility tree because the `title` is empty.
 
````html
<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
	<title> </title>
  	<circle role="none" cx="50" cy="50" r="40" fill="yellow"></circle>
</svg>